samcli/lib/utils/resource_trigger.py

```diff
@@ -58,28 +60,40 @@ def get_single_file_path_handler(file_path_str: str) -> PathHandler:
         """
         file_path = Path(file_path_str).resolve()
         folder_path = file_path.parent
+        case_sensitive = platform.system().lower() == "windows"
         file_handler = RegexMatchingEventHandler(
-            regexes=[f"^{re.escape(str(file_path))}$"], ignore_regexes=[], ignore_directories=True, case_sensitive=True
+            regexes=[f"^{re.escape(str(file_path))}$"],
+            ignore_regexes=[],
+            ignore_directories=True,
+            case_sensitive=case_sensitive,
         )
         return PathHandler(path=folder_path, event_handler=file_handler, recursive=False)
 
     @staticmethod
-    def get_dir_path_handler(dir_path_str: str) -> PathHandler:
+    def get_dir_path_handler(dir_path_str: str, ignore_patterns: Optional[List[str]] = None) -> PathHandler:
         """Get PathHandler for watching a single directory
 
         Parameters
         ----------
         dir_path_str : str
             Folder path in string
+        ignore_patterns : List[str]
+            List of patterns for sub folders/files that should be ignored
 
         Returns
         -------
         PathHandler
             The PathHandler for the folder specified
         """
         dir_path = Path(dir_path_str).resolve()
+
+        # watchdog uses PurePath.match() for patterns
+        # It chooses either PurePosixPath or PureWindowsPath depending on case_sensitive
+        # We have to use the correct Path object in order for path.match() to work properly
+        case_sensitive = platform.system().lower() == "windows"
+
         file_handler = PatternMatchingEventHandler(
-            patterns=["*"], ignore_patterns=[], ignore_directories=False, case_sensitive=True
+            patterns=["*"], ignore_patterns=ignore_patterns, ignore_directories=False, case_sensitive=case_sensitive
```

Comment:
This was `True` by default but now it is only `True` if that is a windows environment. Linux is also case sensitive, is that going to create issues there?

Reply:
Fixed. Should be True for Unix and False for Windows.